src/windows/win-toggle-safe-mode.ps1

```diff
@@ -0,0 +1,153 @@
+. .\src\windows\common\setup\init.ps1
+. .\src\windows\common\helpers\Get-Disk-Partitions.ps1
+
+#########################################################################################################
+# Azure VMs do not natively support Safe Mode because RDP access is disabled in Safe Mode. Some users
+# need to boot their VM in Safe Mode for specific reasons (e.g. uninstalling certain software). Other
+# users may find their VM booting into Safe Mode inadvertantly due to user error or misconfiguration,
+# which will disable RDP access until corrected. This script utilizes the az vm repair extension to 
+# clone the VM into a Hyper-V environment using Nested Virtualization and toggle Safe Boot. The user
+# may then access their VM in Safe Mode via the Rescue VM or revert Safe Mode on their Azure VM. They 
+# may then swap the disk using the `az vm repair restore` functionality.
+#
+# Testing:
+# 1. Copied scripts to newly created Windows Server 2019 Datacenter (Gen 1)
+# 2. Ran win-enable-nested-hyperv once to install Hyper-V, restarted, and ran again to create new nested VM
+# 3. Ran win-toggle-safe-mode.ps1, worked successfully in toggling Safe Mode
+# 4. Set up new VM and ran the following from my local machine, worked successfully (~69 seconds): 
+#    az vm repair run -g sourcevm_group -n sourcevm --custom-script-file .\win-toggle-safe-mode.ps1 --verbose --run-on-repair
+# 5. Tried on a WS 2016 Gen 2 Azure VM, but was unsuccessful, not compatible with Gen 2 right now
+#
+# https://docs.microsoft.com/en-us/cli/azure/ext/vm-repair/vm/repair?view=azure-cli-latest
+# https://docs.microsoft.com/en-us/troubleshoot/azure/virtual-machines/troubleshoot-rdp-safe-mode
+#########################################################################################################
+
+# Initialize script log variables
+$scriptStartTime = get-date -f yyyyMMddHHmmss
+$scriptName = (split-path -path $MyInvocation.MyCommand.Path -leaf).Split('.')[0]
+
+# Initialize script log
+$logFile = "$env:PUBLIC\Desktop\$($scriptName).log"
```

Comment:
Is there any reason you are specifying the log location here? If this is run using 'az vm repair run' to run this script, it should be logged as a file automatically.

Reply:
Here is the run driver which runs this script: https://github.com/Azure/azure-cli-extensions/blob/master/src/vm-repair/azext_vm_repair/scripts/win-run-driver.ps1 

and AZ CLI will output to the user the location of the script as well

Reply:
@swbae31 I honestly copied the style of win-enable-nested-hyperv to output the log to the desktop for convenience! This can be removed 😁

Example of log:
```
[Output 03/25/2021 00:14:59]START: Running script win-toggle-safe-mode
20210325001459
[Info 03/25/2021 00:14:59]#01 - Stopping nested guest VM ProblemVM
[Info 03/25/2021 00:15:08]#02 - Bringing disk online
[Info 03/25/2021 00:15:12]#03 - enumerate partitions for boot config
[Info 03/25/2021 00:15:12]#04 - Checking safeboot flag for E:\boot\bcd
[Info 03/25/2021 00:15:12]#05 - Configuring safeboot flag for E:\boot\bcd
[Info 03/25/2021 00:15:12]#06 - Bringing disk offline
[Output 03/25/2021 00:15:13]#07 - Starting VM
[Output 03/25/2021 00:15:13]END: Please verify status of Safe Mode using MSCONFIG.exe
20210325001513
```
![image](https://user-images.githubusercontent.com/15747450/112403551-fd000080-8ce4-11eb-89db-7e10b93e5844.png)

Inspiration:
https://github.com/Azure/repair-script-library/blob/50556e99fbc624f52f553faeab1c08a89068271c/src/windows/win-enable-nested-hyperv.ps1#L3-L10

Reply:
Removed! I think you were right on this, it looks better imo:
![image](https://user-images.githubusercontent.com/15747450/112407371-5a4b8000-8cec-11eb-8e26-de459f4ebbbe.png)
![image](https://user-images.githubusercontent.com/15747450/112407989-5f5cff00-8ced-11eb-8f84-62c91e4c6cae.png)